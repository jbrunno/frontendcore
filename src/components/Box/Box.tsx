import React from 'react';
import { Box as MuiBox, BoxProps as MuiBoxProps } from '@mui/material';

export type BoxProps = MuiBoxProps;

export function Box(props: BoxProps) {
  return <MuiBox {...props} />;
}

export default Box;
