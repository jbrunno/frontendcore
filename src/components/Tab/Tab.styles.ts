import { StyledComponent } from '@emotion/styled';
import { Tab as MuiTab } from '@mui/material';
import { styled } from '@mui/material/styles';
import { TabProps } from './Tab';

export type TabStyledProps = {
  textTransform?:
    | 'none'
    | 'capitalize'
    | 'uppercase'
    | 'lowercase'
    | 'initial'
    | 'inherit';
};

export const TabStyled: StyledComponent<TabProps> = styled(
  MuiTab,
)<TabStyledProps>`
  text-transform: ${({ textTransform }) => textTransform || 'none'};
`;
