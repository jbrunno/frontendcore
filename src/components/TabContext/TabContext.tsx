import React from 'react';
import {
  TabContext as TabContextMui,
  TabContextProps as MuiTabContextProps,
} from '@mui/lab';

export type TabContextProps = MuiTabContextProps;

export function TabContext(props: TabContextProps) {
  return <TabContextMui {...props} />;
}

export default TabContext;
