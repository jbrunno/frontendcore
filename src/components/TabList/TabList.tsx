import React from 'react';
import {
  TabList as TabListMui,
  TabListProps as MuiTabListProps,
} from '@mui/lab';

export type TabListProps = MuiTabListProps;

export function TabList(props: TabListProps) {
  return <TabListMui {...props} />;
}

export default TabList;
