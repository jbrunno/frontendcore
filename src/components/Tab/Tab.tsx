import React from 'react';
import { TabProps as MuiTabProps } from '@mui/material';
import { TabStyled, TabStyledProps } from './Tab.styles';

export type TabProps = MuiTabProps & TabStyledProps;

export function Tab(props: TabProps) {
  return <TabStyled {...props} />;
}

export default Tab;
