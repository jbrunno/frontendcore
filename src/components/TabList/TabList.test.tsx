import React from 'react';
import { render } from '@test-utils/component-testing';
import TabList from './TabList';
import Tab from '../Tab/Tab';
import TabContext from '../TabContext/TabContext';
import Box from '../Box/Box';

test('renders the TabList', () => {
  const { container } = render(
    <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
      <TabContext value="1">
        <TabList>
          <Tab label="Item One" value="1" />
        </TabList>
      </TabContext>
    </Box>,
  );

  container.querySelectorAll('button').forEach((button) => {
    button.removeAttribute('id');
    button.removeAttribute('aria-controls');
  });

  expect(container).toMatchSnapshot();
});
