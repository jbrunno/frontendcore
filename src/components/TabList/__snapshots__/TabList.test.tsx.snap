// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the TabList 1`] = `
.emotion-0 {
  border-bottom: 1px solid;
  border-color: #0000001F;
}

.emotion-1 {
  overflow: hidden;
  min-height: 48px;
  -webkit-overflow-scrolling: touch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width:599.95px) {
  .emotion-1 .MuiTabs-scrollButtons {
    display: none;
  }
}

.emotion-2 {
  position: relative;
  display: inline-block;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  white-space: nowrap;
  overflow-x: hidden;
  width: 100%;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: Lato,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.25;
  text-transform: uppercase;
  max-width: 360px;
  min-width: 90px;
  position: relative;
  min-height: 48px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 12px 16px;
  overflow: hidden;
  white-space: normal;
  text-align: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: #3B3B4199;
  text-transform: none;
}

.emotion-5::-moz-focus-inner {
  border-style: none;
}

.emotion-5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-5 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-5.Mui-selected {
  color: #4A31FF;
}

.emotion-5.Mui-disabled {
  color: #3B3B4161;
}

.emotion-6 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-7 {
  position: absolute;
  height: 2px;
  bottom: 0;
  width: 100%;
  -webkit-transition: all 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: all 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  background-color: #4A31FF;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <div
      class="MuiTabs-root emotion-1"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed emotion-2"
        style="overflow: hidden; margin-bottom: 0px;"
      >
        <div
          class="MuiTabs-flexContainer emotion-3"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected emotion-4 emotion-5"
            role="tab"
            tabindex="0"
            type="button"
          >
            Item One
            <span
              class="MuiTouchRipple-root emotion-6"
            />
          </button>
        </div>
        <span
          class="MuiTabs-indicator emotion-7"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
  </div>
</div>
`;
